src/classes/RD2_DataMigration_TEST.cls

```diff
@@ -878,6 +878,80 @@ private class RD2_DataMigration_TEST {
         assertBatchSummaryFailed(jobId, 1);
     }
 
+    /**
+     * @description Verifies an error is created when store error in error log settings is disabled
+     * and migration is run in dry run mode
+     */
+    @IsTest
+    private static void shouldLogDryRunErrorWhenStoreErrorIsDisabled() {
+
+        setStoreErrorsInErrorSettingsToFalse();
+
+        npe03__Recurring_Donation__c rd = getLegacyRecurringDonationBuilder(getContact().Id)
+            .withId(null)
+            .build();
+        insert rd;
+
+        //Make sure an error is generated when dry run is run
+        TDTM_ProcessControl.setRecursionFlag(TDTM_ProcessControl.flag.RD, true);
+        rd.npe03__Contact__c = null;
+        update rd;
+        TDTM_ProcessControl.setRecursionFlag(TDTM_ProcessControl.flag.RD, false);
+
+        Id jobId = executeAndFinishDryRun();
+
+        List<Error__c> errors = errorGateway.getRecords();
+        System.assertEquals(1, errors.size(), 'Validation error should be logged');
+
+        System.assertEquals(true, errors[0].Full_Message__c.contains(System.Label.RD_DonorIsRequired), 'Error message should match');
```

Comment:
Let's extract common code verifying `RD_DonorIsRequired` error into a new helper method and call it from three tests verifying donor setup.